### ISSUE TITLE
Out-of-Distribution Detection Using Layerwise Uncertainty in Deep Neural Networks

### ISSUE THREAD

**Issue by jo-kwsm** [jo-kwsm/paper_summary#32]
# INFO
## author
Hirono Okamoto, Masahiro Suzuki, Yutaka Matsuo

## affiliation
University of Tokyo

## conference or year
2020

## link
[openreview](https://openreview.net/forum?id=rklVOnNtwH)
[実装](https://github.com/unknown-ai/UFEL)

# 概要
softmax はどのようなデータにも高い尤度を割り当ててしまい、OOD 検出が困難なことが知られている。そこで、本論文では確信度を同時に算出する手法を提案した。
![Screenshot from 2021-02-03 23-27-14](https://user-images.githubusercontent.com/54238261/106761201-ebe92a80-6677-11eb-8943-899099128c70.png)

# 提案手法
提案するmodel を以下に示す。中間層で平均と分散を算出しながら確率的に伝播させていくことで擬似的に確信度を算出する。
![Screenshot from 2021-02-03 23-28-15](https://user-images.githubusercontent.com/54238261/106761686-6f0a8080-6678-11eb-8b28-d59ee235ad0d.png)

各層ごとに算出される確信度を以下の2つの方法で合計し、最終的な確信度を算出する。
![Screenshot from 2021-02-03 23-28-27](https://user-images.githubusercontent.com/54238261/106761710-7467cb00-6678-11eb-9f93-964d392442a5.png)
![Screenshot from 2021-02-03 23-29-53](https://user-images.githubusercontent.com/54238261/106761724-7893e880-6678-11eb-97f5-b003d9970422.png)

# 検証
OOD 検出精度の比較。既存のモデルよりも強い。
![Screenshot from 2021-02-03 23-30-11](https://user-images.githubusercontent.com/54238261/106761736-7af64280-6678-11eb-863a-f8c8432a162f.png)

D<sub>validation</sub> にOOD を用いているが、実際にはそのようなデータが用意できないことも多い。そのため、validation 用のOOD を使用しなかった場合の精度を比較。
![Screenshot from 2021-02-03 23-30-22](https://user-images.githubusercontent.com/54238261/106761744-7cc00600-6678-11eb-9841-8cd1d46b6d9c.png)

# 新規性
- NN 内で平均と分散を算出することでOOD 検出を実施

# 議論，展望

# Comment
- ベイズ深層学習がこんな感じだった気がする。勉強したい。

# date
Feb 3, 2021
